scipy/signal/spectral.py

```diff
@@ -766,11 +770,15 @@ def _spectral_helper(x, y, fs=1.0, window='hann', nperseg=256,
                 pad_shape[-1] = x.shape[-1] - y.shape[-1]
                 y = np.concatenate((y, np.zeros(pad_shape)), -1)
 
+    # parse window; if array like, then set nperseg = win.shape
+    win, nperseg = _triage_segments(window, nperseg)              
+        
     # X and Y are same length now, can test nperseg with either
     if x.shape[-1] < nperseg:
         warnings.warn('nperseg = {0:d}, is greater than input length = {1:d}, '
                       'using nperseg = {1:d}'.format(nperseg, x.shape[-1]))
         nperseg = x.shape[-1]
+        win, nperseg = _triage_segments(window, nperseg) 
```

Comment:
I'm not quite clear what you mean by "when ```nperseg==None``` and the window has greater length than the signal".
As currently written, ```_spectral_helper``` checks whether nperseg is greater than the length of the signal, gives a warning, and then sets nperseg to the length of the window. This is in lines 780-784. In line 785 I added another call to ```_triage_segments``` so that it gets a new window corresponding to the new value of nperseg.

There's several tests in test_spectral that seem to be verifying whether this happens, e..g., ```TestCSD.test_short_data``` so I figured I should stay consistent with that behavior.